src/Cool/String/test.cpp

```diff
@@ -27,43 +80,40 @@ TEST_CASE("[Cool::String] replace()")
         CHECK(Cool::String::find_replacement("hello", replacements).value() == "1");
         CHECK(Cool::String::find_replacement("hell", replacements) == std::nullopt);
     }
-}
-
-TEST_CASE("replace_all()")
-{
-    std::string text = "Effect_intensity";
-    Cool::String::replace_all(text, "_", " ");
-    CHECK(text == "Effect intensity");
-}
-
-TEST_CASE("[Cool::String] replace_at()")
-{
-    CHECK(Cool::String::replace_at(3, 5, "0123456789", "abcde") == "012abcde56789");
+    SUBCASE("[Cool::String] replace_at()")
+    {
+        CHECK(Cool::String::replace_at(3, 5, "0123456789", "abcde") == "012abcde56789");
+    }
 }
 
 TEST_CASE("[Cool::String] find_matching_pair()")
 {
     SUBCASE("When there is nested delimiters")
     {
-        CHECK(Cool::String::find_matching_pair("abc(de(fg)h)ij") == std::make_pair<size_t, size_t>(3, 11));
-        CHECK(Cool::String::find_matching_pair("abc(de(fgh)ij") == std::nullopt);
+        CHECK(Cool::String::find_matching_pair({"abc(de(fg)h)ij"}) == std::make_pair<size_t, size_t>(3, 11));
+        CHECK(Cool::String::find_matching_pair({"abc(de(fgh)ij"}) == std::nullopt);
     }
     SUBCASE("When the delimiters are the first and last characters of the text")
     {
-        CHECK(Cool::String::find_matching_pair("(abc)") == std::make_pair<size_t, size_t>(0, 4));
+        CHECK(Cool::String::find_matching_pair({"(abc)"}) == std::make_pair<size_t, size_t>(0, 4));
     }
     SUBCASE("When the text is empty")
     {
-        CHECK(Cool::String::find_matching_pair("") == std::nullopt);
+        CHECK(Cool::String::find_matching_pair({""}) == std::nullopt);
     }
     SUBCASE("When the first opening delimiter is at the end of the text")
     {
-        CHECK(Cool::String::find_matching_pair("abc(") == std::nullopt);
+        CHECK(Cool::String::find_matching_pair({"abc("}) == std::nullopt);
     }
     SUBCASE("When the opening and closing delimiters are the same")
     {
-        CHECK(Cool::String::find_matching_pair("ab|cde|fg", '|', '|').value() == std::make_pair<size_t, size_t>(2, 6));
-        CHECK(Cool::String::find_matching_pair("ab|cde|fg|hi", '|', '|').value() == std::make_pair<size_t, size_t>(2, 6));
+        CHECK(Cool::String::find_matching_pair({.text = "ab|cde|fg", .opening = '|', .closing = '|'}).value() == std::make_pair<size_t, size_t>(2, 6));
+        CHECK(Cool::String::find_matching_pair({.text = "ab|cde|fg|hi", .opening = '|', .closing = '|'}).value() == std::make_pair<size_t, size_t>(2, 6));
+    }
+    SUBCASE("Test offset parameter")
+    {
+        CHECK(Cool::String::find_matching_pair({"fes(svrsnv)", 2, '(', ')'}).value() == std::make_pair<size_t, size_t>(3, 10));

```

Comment:
Test a string with two sets a parentheses and an offset such that the function finds the second pair of parentheses